src/main/java/org/tron/program/KeystoreFactory.java

```diff
@@ -0,0 +1,153 @@
+package org.tron.program;
+
+import com.beust.jcommander.JCommander;
+import java.io.File;
+import java.io.IOException;
+import java.util.Scanner;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.tron.common.crypto.ECKey;
+import org.tron.common.utils.ByteArray;
+import org.tron.common.utils.Utils;
+import org.tron.core.Constant;
+import org.tron.core.config.args.Args;
+import org.tron.keystore.CipherException;
+import org.tron.keystore.Credentials;
+import org.tron.keystore.WalletUtils;
+
+
+
+
+public class KeystoreFactory {
+
+  private static final Logger logger = LoggerFactory.getLogger("KeystoreFactory");
+  private static final String FilePath = "Wallet";
+
+  private boolean priKeyValid(String priKey) {
+    if (StringUtils.isEmpty(priKey)) {
+      logger.warn("Warning: PrivateKey is empty !!");
+      return false;
+    }
+    if (priKey.length() != 64) {
+      logger.warn("Warning: PrivateKey length need 64 but " + priKey.length() + " !!");
+      return false;
+    }
+    //Other rule;
+    return true;
+  }
+
+  private void genKeystore() throws CipherException, IOException {
+    String password = WalletUtils.inputPassword2Twice();
+
+    ECKey eCkey = new ECKey(Utils.random);
+    File file = new File(FilePath);
+    if (!file.exists()) {
+      file.mkdir();
+    } else {
+      if (!file.isDirectory()) {
+        file.delete();
+        file.mkdir();
+      }
+    }
+    String fileName = WalletUtils.generateWalletFile(password, eCkey, file, true);
+    System.out.println("Gen a keystore its name " + fileName);
+    Credentials credentials = WalletUtils.loadCredentials(password, new File(file, fileName));
+    System.out.println("Your address is " + credentials.getAddress());;
+  }
+
+  private void importPrivatekey() throws CipherException, IOException {
+    Scanner in = new Scanner(System.in);
+    String privateKey;
+    System.out.println("Please input private key.");
+    while (true) {
+      String input = in.nextLine().trim();
+      privateKey = input.split("\\s+")[0];
+      if (priKeyValid(privateKey)){
+        break;
+      }
+      System.out.println("Invalid private key, please input again.");
+    }
+
+    String password = WalletUtils.inputPassword2Twice();
+
+    ECKey eCkey = ECKey.fromPrivate(ByteArray.fromHexString(privateKey));
+    File file = new File(FilePath);
+    if (!file.exists()) {
+      file.mkdir();
+    } else {
+      if (!file.isDirectory()) {
+        file.delete();
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Do something with the "boolean" value returned by "delete". ([squid:S899](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS899))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=938&issues=AWPK23iJkDisRwWBfuxF&open=AWPK23iJkDisRwWBfuxF)

[ISSUE_KEY]: AWPK23iJkDisRwWBfuxF